tpu_commons/utils.py

```diff
@@ -85,11 +85,45 @@ def hbm_usage_bytes(devices: Any) -> List[Tuple[int, int]]:
     return usage
 
 
+def get_device_name(num_devices: int | None = None):
+    kind = jax.devices()[0].device_kind
+    if 'TPU' not in kind:
+        raise RuntimeError('Expected TPU devices')
+    suffix = ''
+    if kind.endswith(' lite'):
+        kind = kind[:-len(' lite')]
+        suffix = 'e'
+    if kind.endswith('p'):
+        kind = kind[:-1]
+        suffix = 'p'
+    if kind == 'TPU7x':
+        kind = 'TPU v7'
+    assert kind[:-1] == 'TPU v', kind
+    kind += suffix
+    if num_devices is not None:
+        kind += f'-{num_devices}'
+    return kind
+
+
+def get_device_hbm_limit() -> int:
+
+    device_kind = get_device_name()
+    if device_kind == "TPU v5p" or device_kind == "TPU v5":
+        return 95 * GBYTES
+    elif device_kind == "TPU v5e":
+        return 16 * GBYTES
+    elif device_kind == "TPU v6e" or device_kind == "TPU v4":
+        return 32 * GBYTES
+    elif device_kind == "TPU v7":
+        return 192 * GBYTES
```

Comment:
I think this should be 192 * GBYTES / 2 because each jax device (v7x core) has 1/2 of the total chip HBM.

@wenxindongwork